### ISSUE TITLE
V4.0.0 complains about S3 permissions

### ISSUE THREAD

**Issue by stevehodgkiss** [logstash-plugins/logstash-output-s3#110]
Version 4.0.0 of the plugin complains about S3 permission issues, whereas 3.2.0 does not (and works). To see if it was a missing permission I tried granting `s3*` to the bucket (via the AWS role of the box) but that didn't fix it. Reverted to 3.2.0 and S3 uploads worked again.

- Version: 4.0.0
- Operating System: ubuntu 16.04

Log entry:

```
{:timestamp=>"2017-01-03T00:27:44.601000+0000", :message=>"Pipeline aborted due to error", :exception=>"LogStash::ConfigurationError", :error=>"Logstash must have the privileges to write to root bucket `[snip]`, check you credentials or your permissions.", :backtrace=>["/opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-output-s3-4.0.0/lib/logstash/outputs/s3.rb:187:in `register'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-core-2.4.1-java/lib/logstash/output_delegator.rb:75:in `register'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-core-2.4.1-java/lib/logstash/pipeline.rb:181:in `start_workers'", "org/jruby/RubyArray.java:1613:in `each'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-core-2.4.1-java/lib/logstash/pipeline.rb:181:in `start_workers'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-core-2.4.1-java/lib/logstash/pipeline.rb:136:in `run'", "/opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-core-2.4.1-java/lib/logstash/agent.rb:491:in `start_pipeline'"], :level=>:error}
```

Config:

```
output {
     s3 {
         codec => "json_lines"
         bucket => "[snip]"
         size_file => 512000
         time_file => 5
         encoding => "gzip"
     }
}
```

**Comment by ph** [comment]
You are using AWS Role and you aren't settings up anything in the environment? Correct?

You can disable the check with `validate_credentials_on_root_bucket => false` in your configuration.

**Comment by andrewvc** [comment]
We'll need more info on this, but our current error logging isn't up to snuff because we swallow the exception here: https://github.com/logstash-plugins/logstash-output-s3/blob/dbc474bd54b64ea985ea59f70777d6d13c5be211/lib/logstash/outputs/s3/write_bucket_permission_validator.rb#L14 I'll open a PR to fix that so we can diagnose your issue better.

**Comment by ph** [comment]
fixed with #112
